src/main/java/com/hivemq/mqtt/handler/OutboundInterceptorHandler.java

```diff
@@ -0,0 +1,95 @@
+package com.hivemq.mqtt.handler;
+
+import com.hivemq.extension.sdk.api.annotations.NotNull;
+import com.hivemq.extensions.handler.*;
+import com.hivemq.mqtt.message.PINGRESP;
+import com.hivemq.mqtt.message.connack.CONNACK;
+import com.hivemq.mqtt.message.disconnect.DISCONNECT;
+import com.hivemq.mqtt.message.puback.PUBACK;
+import com.hivemq.mqtt.message.pubcomp.PUBCOMP;
+import com.hivemq.mqtt.message.publish.PUBLISH;
+import com.hivemq.mqtt.message.pubrec.PUBREC;
+import com.hivemq.mqtt.message.pubrel.PUBREL;
+import com.hivemq.mqtt.message.suback.SUBACK;
+import com.hivemq.mqtt.message.unsuback.UNSUBACK;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelOutboundHandlerAdapter;
+import io.netty.channel.ChannelPromise;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * @author Daniel Krüger
+ */
+@Singleton
+@ChannelHandler.Sharable
+public class OutboundInterceptorHandler extends ChannelOutboundHandlerAdapter {
+
+    private final @NotNull ConnackOutboundInterceptorHandler connackOutboundInterceptorHandler;
+    private final @NotNull PublishOutboundInterceptorHandler publishOutboundInterceptorHandler;
+    private final @NotNull SubackOutboundInterceptorHandler subackOutboundInterceptorHandler;
+    private final @NotNull UnsubackOutboundInterceptorHandler unsubackOutboundInterceptorHandler;
+
+    private final @NotNull PingInterceptorHandler pingInterceptorHandler;
+    private final @NotNull PubackInterceptorHandler pubackInterceptorHandler;
+    private final @NotNull PubrecInterceptorHandler pubrecInterceptorHandler;
+    private final @NotNull PubcompInterceptorHandler pubcompInterceptorHandler;
+    private final @NotNull PubrelInterceptorHandler pubrelInterceptorHandler;
+    private final @NotNull DisconnectInterceptorHandler disconnectInterceptorHandler;
+
+
+    @Inject
+    public OutboundInterceptorHandler(final @NotNull ConnackOutboundInterceptorHandler connackOutboundInterceptorHandler,
```

Comment:
formatting, wrapped arguments should start in a new line (I think we are still missing the formatting rules in the composite)

Reply:
added the formatting rules to the repo